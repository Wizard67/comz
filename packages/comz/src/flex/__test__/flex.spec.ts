import { mount } from '@vue/test-utils'
import Flex from '../flex.vue'

describe('Flex.vue', () => {
  describe('props', () => {
    it('inline', () => {
      const wrapper = mount(Flex, {
        props: {
          inline: true
        }
      })

      expect(wrapper.classes()).toContain('cflex--inline')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('gap', () => {
      const wrapper = mount(Flex, {
        props: {
          gap: '10px'
        }
      })

      expect(wrapper.attributes('style')).toContain('--cflex-gap: 10px;')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('justify', () => {
      const wrapper = mount(Flex, {
        props: {
          justify: 'center'
        }
      })

      expect(wrapper.attributes('style')).toContain('--cflex-justify: center;')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('align', () => {
      const wrapper = mount(Flex, {
        props: {
          align: 'flex-end'
        }
      })

      expect(wrapper.attributes('style')).toContain('--cflex-align: flex-end;')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('center', () => {
      const wrapper = mount(Flex, {
        props: {
          center: true
        }
      })

      expect(wrapper.classes()).toContain('cflex--center')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('vertical', () => {
      const wrapper = mount(Flex, {
        props: {
          vertical: true
        }
      })

      expect(wrapper.classes()).toContain('cflex--vertical')
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('slots', () => {
    it('default', () => {
      const content = "flex"

      const wrapper = mount(Flex, {
        slots: {
          default: content
        }
      })

      expect(wrapper.text()).toContain(content)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
