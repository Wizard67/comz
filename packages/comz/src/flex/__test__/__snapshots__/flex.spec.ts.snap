// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue props align 1`] = `
<section
  class="cflex"
  style="--cflex-gap: initial; --cflex-align: flex-end; --cflex-justify: initial;"
>
  
  
</section>
`;

exports[`Button.vue props center 1`] = `
<section
  class="cflex cflex--center"
  style="--cflex-gap: initial; --cflex-align: initial; --cflex-justify: initial;"
>
  
  
</section>
`;

exports[`Button.vue props fill 1`] = `
<section
  class="cflex cflex--fill"
  style="--cflex-gap: initial; --cflex-align: initial; --cflex-justify: initial;"
>
  
  
</section>
`;

exports[`Button.vue props gap 1`] = `
<section
  class="cflex"
  style="--cflex-gap: 10px; --cflex-align: initial; --cflex-justify: initial;"
>
  
  
</section>
`;

exports[`Button.vue props inline 1`] = `
<section
  class="cflex cflex--inline"
  style="--cflex-gap: initial; --cflex-align: initial; --cflex-justify: initial;"
>
  
  
</section>
`;

exports[`Button.vue props justify 1`] = `
<section
  class="cflex"
  style="--cflex-gap: initial; --cflex-align: initial; --cflex-justify: center;"
>
  
  
</section>
`;

exports[`Button.vue props vertical 1`] = `
<section
  class="cflex cflex--vertical"
  style="--cflex-gap: initial; --cflex-align: initial; --cflex-justify: initial;"
>
  
  
</section>
`;

exports[`Button.vue slots default 1`] = `
<section
  class="cflex"
  style="--cflex-gap: initial; --cflex-align: initial; --cflex-justify: initial;"
>
  
  
  flex
  
  
</section>
`;
